capreolus/extractor/bagofwords.py

```diff
@@ -0,0 +1,172 @@
+import pickle
+import os
+import time
+
+from capreolus.extractor import Extractor
+from capreolus.tokenizer import Tokenizer
+from capreolus.utils.loginit import get_logger
+from tqdm import tqdm
+import numpy as np
+from collections import Counter, defaultdict
+
+from capreolus.registry import Dependency
+
+logger = get_logger(__name__)  # pylint: disable=invalid-name
+
+
+class BagOfWords(Extractor):
+    """ Bag of Words (or bag of trigrams when `datamode=trigram`) extractor. Used with the DSSM reranker. """
+
+    name = "bagofwords"
+    dependencies = {
+        "index": Dependency(module="index", name="anserini", config_overrides={"indexstops": True, "stemmer": "none"}),
+        "tokenizer": Dependency(module="tokenizer", name="anserini"),
+    }
+    pad = 0
+    pad_tok = "<pad>"
+    tokenizer_name = "anserini"
+
+    @staticmethod
+    def config():
+        datamode = "unigram"  # type of input: 'unigram' or 'trigram'
+        keepstops = False  # include stopwords in the reranker's input
```

Comment:
I think this isn't used -- it was before the rewrite, but now it's a config option of the tokenizer dependency